app/next-client-app/api/files.ts

```diff
@@ -37,8 +38,8 @@ export async function requestFile(
         file_type: file_type,
       }),
     });
-    return { success: true };
   } catch (error: any) {
     return { success: false, errorMessage: error.message };
   }
+  redirect(`/scanreports/${scan_report_id}/downloads`);
```

Comment:
We shouldn't do redirection in the `api` fetching code, this should be just responsible for fetching. Wherever is calling these functions should  do the redirection (I remember that we have broken this rule somewhere else that needs fixing, but lets not duplicate it)

Reply:
I see. Redirection happened in the updating SR tables details as well. But the thing is without doing redirect here, after users request the file successfully, and being pushed to `downloads` page, the page is not revalidated and users can't see the "In-progress" indication intermediately (they need to refresh the `downloads` page to see this).

Reply:
Hmm maybe it's revalidate and then redirect in the calling code on the page? If not - we can come back to improve this pattern later

Reply:
yep, so the pattern is changed now. Revalidate in the `backend` of `NextJS`, then `router.push` on the place calling the code. This changes have applied for both the Table details update and request file downloading.